pkg/sdk/sdk_test.go

```diff
@@ -0,0 +1,503 @@
+	}
+}
+func Test_GeneratePdf(t *testing.T) {
+	tests := []struct {
+		name              string
+		id                string
+		data              map[string]interface{}
+		setupFunc         func() *httptest.Server
+		mockServerHandler func(w http.ResponseWriter, r *http.Request)
+		ValidateFunc      func(b []byte, err error)
+		cleanupFunc       func(*httptest.Server)
+		expectedResponse  model.Response
+	}{
+		{
+			name: "Success:: GeneratePdf",
+			id:   "1",
+			data: map[string]interface{}{"id": "1"},
+			setupFunc: func() *httptest.Server {
+				svr := testServer("/v1/generate/{id}", http.MethodPost, func(w http.ResponseWriter, r *http.Request) {
+					vars := mux.Vars(r)
+					//we take id as a parameter from url path
+					id, ok := vars["id"]
+					if !ok {
+						response.ToJson(w, http.StatusBadRequest, "id not found", nil)
+						return
+					}
+					var data GenerateReq
+					err := json.NewDecoder(r.Body).Decode(&data)
+					if err != nil {
+						response.ToJson(w, http.StatusBadRequest, "error decoding data", nil)
+						log.Error(err.Error())
+						return
+					}
+					data.Id = id
+					if err != nil {
+						response.ToJson(w, http.StatusBadRequest, "file read fail", nil)
+						log.Error(err.Error())
+						return
+					}
+
+					if !reflect.DeepEqual(data.Values["id"], "1") {
+						t.Errorf("Want: %v, Got: %v", "1", data.Values["id"])
+					}
+					if !reflect.DeepEqual(r.Header.Get("Content-Type"), "application/json") {
+						t.Errorf("Want: %v, Got: %v", "application/json", r.Header.Get("Content-Type"))
+					}
+					response.ToJson(w, http.StatusOK, "", nil)
+				})
+				return svr
+			},
+			ValidateFunc: func(b []byte, err error) {
+				if err != nil {
+					t.Errorf("Want: %v, Got: %v", nil, err.Error())
+				}
+				var tempResp model.Response
+				err = json.Unmarshal(b, &tempResp)
+				if err != nil {
+					t.Error(err)
+					return
+				}
+				if !reflect.DeepEqual(tempResp, model.Response{
+					Status:  http.StatusOK,
+					Message: "",
+					Data:    nil,
+				}) {
+					t.Errorf("Want: %v, Got: %v", model.Response{
+						Status:  http.StatusOK,
+						Message: "",
+						Data:    nil,
+					}, tempResp)
+				}
+			},
+			cleanupFunc: func(svr *httptest.Server) {
+				svr.Close()
+			},
+		},
+		{
+			name: "Failure:: GeneratePdf :: incorrect status code",
+			id:   "1",
+			data: map[string]interface{}{"id": "1"},
+			setupFunc: func() *httptest.Server {
+				svr := testServer("/v1/generate/{id}", http.MethodPost, func(w http.ResponseWriter, r *http.Request) {
+					vars := mux.Vars(r)
+					//we take id as a parameter from url path
+					id, ok := vars["id"]
+					if !ok {
+						response.ToJson(w, http.StatusBadRequest, "id not found", nil)
+						return
+					}
+					var data GenerateReq
+					err := json.NewDecoder(r.Body).Decode(&data)
+					if err != nil {
+						response.ToJson(w, http.StatusBadRequest, "error decoding data", nil)
+						log.Error(err.Error())
+						return
+					}
+					data.Id = id
```

Comment:
What's the purpose of performing the above logic if they are not to be used?

Reply:
Done